Doc/tools/.nitignore

```diff
@@ -0,0 +1,602 @@
+# All RST files under Doc/ -- except these -- must pass Sphinx nit-picky mode,
+# as tested on the CI via touch-clean-files.py in doc.yml.
+# Add blank lines between files and keep them sorted lexicographically
+# to help avoid merge conflicts.
+
+Doc/c-api/allocation.rst
+
+Doc/c-api/apiabiversion.rst
+
+Doc/c-api/arg.rst
+
+Doc/c-api/bool.rst
+
```

Comment:
What's the point to insert an empty line after each rule line?

Reply:
To reduce merge conflicts.

(Over at typeshed, we've rarely run into merge conflicts for our similar [stubtest allowlists](https://github.com/python/typeshed/blob/main/tests/stubtest_allowlists/py3_common.txt), and they've been very easy to solve when we have run into them. So I, personally, find this somewhat unnecessary. But I don't feel strongly about it.)

Reply:
Yeah, and the newlines also makes it harder to sort the lines using a text editor. Maybe I'll remove them and we can add them back in the future if it turns into a common problem.

Reply:
Yeah, the changes of that are substantially reduced now that the full list is here, and automatic sort would be nice.